src/TrackTraversingAlgorithms.cpp

```diff
@@ -29,6 +170,97 @@ void VolumeCalculator::execute() {
 
 
 /**
+ * @brief Constructor for CentroidGenerator calls the TraverseTracks
+ *        constructor and imports refernces to both the FSR volumes and FSR
+ *        locks arrays
+ * @param track_generator The TrackGenerator to pull tracking information from
+ */
+CentroidGenerator::CentroidGenerator(TrackGenerator* track_generator)
+                                  : TraverseTracks(track_generator) {
+
+  _FSR_volumes = track_generator->getFSRVolumes();
+  _FSR_locks = track_generator->getFSRLocks();
+  _quadrature = track_generator->getQuadrature();
+}
+
+
+/**
+ * @brief Calculates the centroid of every FSR
+ * @details On each segment, onTrack(...) calculates the contribution to each
+ *          FSR centroid and saves the centroids in the centroids array
+ *          provided by setCentroids(...).
+ */
+void CentroidGenerator::execute() {
+#pragma omp parallel
+  {
+    loopOverTracks(NULL);
+  }
+}
+
+
+/**
+ * @brief Specifies an array to save calculated FSR centroids
+ * @brief centroids The array of FSR centroids pointers
+ */
+void CentroidGenerator::setCentroids(Point** centroids) {
+  _centroids = centroids;
+}
+
+
+/**
+ * @brief Centroid contributions are calculated for every segment in the Track
+ * @param track The Track associated with the segments
+ * @param segments The segments whose contributions are added to the centroids
+ */
+void CentroidGenerator::onTrack(Track* track, segment* segments) {
+
+  /* Extract common information from the Track */
+  Point* start = track->getStart();
+  int azim_index = track->getAzimIndex();
+  double phi = track->getPhi();
+
+  /* Compute the Track cross-sectional area */
+  double wgt = _quadrature->getAzimSpacing(azim_index)
+      * _quadrature->getAzimWeight(azim_index);
```

Comment:
I don't think the comment is fitting here. I would just say that you are computing the track azimuthal weight.